apis/python/src/tiledbsoma/_tdb_handles.py

```diff
@@ -419,6 +409,64 @@ def enum(self, label: str) -> tiledb.Enumeration:
         raise NotImplementedError
 
 
+class DataFrameWrapper(SOMAArrayWrapper, Wrapper[clib.SOMADataFrame]):
+    """Wrapper around a Pybind11 SOMADataFrame handle."""
+
+    @classmethod
+    def _opener(
+        cls,
+        uri: str,
+        mode: options.OpenMode,
+        context: SOMATileDBContext,
+        timestamp: int,
+    ) -> clib.SOMADataFrame:
+        open_mode = clib.OpenMode.read if mode == "r" else clib.OpenMode.write
+        config = {k: str(v) for k, v in context.tiledb_config.items()}
+        column_names: List[str] = []
+        result_order = clib.ResultOrder.automatic
+        return clib.SOMADataFrame.open(
+            uri,
+            open_mode,
+            config,
+            column_names,
+            result_order,
+            (0, timestamp),
+        )
```

Comment:
It looks like this can be implemented without the need to separate out the codepaths here, since they are identical save for the `return clib.ONLY_THIS_CHANGES.open(` line:

```python
_ArrType = TypeVar("_ArrType", bound=clib.SOMAArray)
class SOMAArrayWrapper(Wrapper[_ArrType]):
  # We don't mark this as a ClassVar when we use this pattern
  # in TileDBObject, which may be a mistake (?)
  _WRAPPED_TYPE: ClassVar[Type[_ArrType]]

  @classmethod
  def _opener(cls, ...) -> _ArrType:
    ...
    return cls._WRAPPED_TYPE.open(...)

class DataFrameWrapper(SOMAArrayWrapper[clib.SOMADataFrame]):
  _WRAPPED_TYPE = clib.SOMADataFrame

  # other methods

class DenseNDArrayWrapper(SOMAArrayWrapper[clib.SOMADenseNDArray]):
  _WRAPPED_TYPE = clib.SOMADenseNDArray

  # other methods
```